hnn_core/cell.py

```diff
@@ -184,27 +184,23 @@ def insert_dipole(self, yscale):
 
         # dends must have already been created!!
         # it's easier to use wholetree here, this includes soma
-        seclist = h.SectionList()
-        seclist.wholetree(sec=self.soma)
-        # create a python section list list_all
-        list_all = [sec for sec in seclist]
-        for sect in list_all:
+        sec_list = h.SectionList()
+        sec_list.wholetree(sec=self.soma)
+        sec_list = [sec for sec in sec_list]
+        for sect in sec_list:
             sect.insert('dipole')
         # Dipole is defined in dipole_pp.mod
-        self.dipole_pp = [h.Dipole(1, sec=sect) for sect in list_all]
+        self.dipole_pp = [h.Dipole(1, sec=sect) for sect in sec_list]
         # setting pointers and ztan values
-        for sect, dpp in zip(list_all, self.dipole_pp):
-            # assign internal resistance values to dipole point process (dpp)
-            dpp.ri = h.ri(1, sec=sect)
+        for sect, dpp in zip(sec_list, self.dipole_pp):
+            dpp.ri = h.ri(1, sec=sect)  # assign internal resistance
             # sets pointers in dipole mod file to the correct locations
-            # h.setpointer(ref, ptr, obj)
-            h.setpointer(sect(0.99)._ref_v, 'pv', dpp)
-            h.setpointer(self.dpl_ref, 'Qtotal', dpp)
+            dpp._ref_pv = sect(0.99)._ref_v
+            dpp._ref_Qtotal = self.dpl_ref
             # gives INTERNAL segments of the section, non-endpoints
             # creating this because need multiple values simultaneously
-            loc = np.array([seg.x for seg in sect])
-            # these are the positions, including 0 but not L
             pos = np.array([seg.x for seg in sect.allseg()])
+            loc = pos[1:-1]  # positions without 0 and L
```

Comment:
Not necessary to change now, but can we use a more descriptive naming scheme? 
Maybe `pos_all` and `pos_internal` when you get to the more invasive changes.

Reply:
done, I was hoping to get rid of one of the variables in the future. But I changed the naming for the interim

Reply:
Yeah I was thinking the same. It would really just amount to awkward indexing further down so at least it's readable now.

I'll go ahead and merge once the CI's are green.

Reply:
wdyt about the last commit? I can revert it if you don't like it?

Reply:
Sorry missed this the question here, thought it was great. Huge fan of reducing indexing tricks as it makes debugging a lot easier down the line.